core/auth/src/main/java/software/amazon/awssdk/auth/credentials/AwsBasicCredentials.java

```diff
@@ -121,4 +147,67 @@ public int hashCode() {
         hashCode = 31 * hashCode + Objects.hashCode(secretAccessKey());
         return hashCode;
     }
+
+    @Override
+    public Builder toBuilder() {
+        return builder().accessKeyId(accessKeyId)
+                        .secretAccessKey(secretAccessKey)
+                        .validateCredentials(validateCredentials)
+                        .provider(provider);
+    }
+
+    @Override
+    public AwsBasicCredentials copy(Consumer<? super AwsBasicCredentials.Builder> modifier) {
+        return ToCopyableBuilder.super.copy(modifier);
+    }
+
+    /**
+     * A builder for creating an instance of {@link AwsBasicCredentials}. This can be created with the static
+     * {@link #builder()} method.
+     */
+    public static final class Builder implements CopyableBuilder<AwsBasicCredentials.Builder, AwsBasicCredentials> {
+        private String accessKeyId;
+        private String secretAccessKey;
+        private String provider;
+        private boolean validateCredentials = true;
+
+        private Builder() {
+        }
+
+        /**
+         * The AWS access key, used to identify the user interacting with services.
+         */
+        public Builder accessKeyId(String accessKeyId) {
+            this.accessKeyId = accessKeyId;
+            return this;
+        }
+
+        /**
+         * The AWS secret access key, used to authenticate the user interacting with services.
+         */
+        public Builder secretAccessKey(String secretAccessKey) {
+            this.secretAccessKey = secretAccessKey;
+            return this;
+        }
+
+        /**
+         * The name of the identity provider that created this credential identity.
+         */
+        public Builder provider(String provider) {
+            this.provider = provider;
+            return this;
+        }
+
+        /**
+         * Whether this class should verify that accessKeyId and secretAccessKey are not null.
+         */
+        public Builder validateCredentials(Boolean validateCredentials) {
```

Comment:
Will this method be publicly available? If it's not beneficial to the public user, can we mark it as @SdkInternalApi?